agents/mysql/perfschema/perfschema.go

```diff
@@ -266,16 +267,18 @@ func makeBuckets(current, prev map[string]*eventsStatementsSummaryByDigest, l *l
 			l.Debugf("Normal query: %s.", currentESS)
 		}
 
+		fingerprint, truncated := truncate.Query(*currentESS.DigestText)
 		count := inc(currentESS.CountStar, prevESS.CountStar)
 		mb := &agentpb.MetricsBucket{
 			Common: &agentpb.MetricsBucket_Common{
 				Schema:                 pointer.GetString(currentESS.SchemaName), // TODO can it be NULL?
 				Queryid:                *currentESS.Digest,
-				Fingerprint:            *currentESS.DigestText,
+				Fingerprint:            fingerprint,
 				NumQueries:             count,
 				NumQueriesWithErrors:   inc(currentESS.SumErrors, prevESS.SumErrors),
 				NumQueriesWithWarnings: inc(currentESS.SumWarnings, prevESS.SumWarnings),
 				AgentType:              inventorypb.AgentType_QAN_MYSQL_PERFSCHEMA_AGENT,
+				IsTruncated:            truncated,
```

Comment:
IsTruncated is documented as "Indicates if query examples is too long and was truncated" (with a grammar error).
The fingerprint is not a query example. Fingerprint also can't contain invalid UTF-8 bytes sequences – there are not allowed in table or columns names, and all values are absent in the fingerprint.

I think we should stay on the topic of this ticket and fix UTF-8 problems where they exist – in query examples.

(that's why I mentioned several times that we need realistic examples – it is not about replacing invalid UTF-8 bytes sequences in opaque strings, but where they are actually can be present)